core/lib/spree/testing_support/factories/payment_factory.rb

```diff
@@ -6,7 +6,7 @@
 FactoryGirl.define do
   factory :payment, aliases: [:credit_card_payment], class: Spree::Payment do
     association(:payment_method, factory: :credit_card_payment_method)
-    association(:source, factory: :credit_card)
+    source { create(:credit_card, user: order.user) }
```

Comment:
I *think* this can be done with the `association` method, which will hit the database less when using `build_stubbed`: 

```
association :source, factory: [:credit_card, user: order.user]
```